mne/epochs.py

```diff
@@ -2072,7 +2105,57 @@ def read_epochs(fname, proj=True, add_eeg_ref=True, verbose=None):
         epochs.selection = selection
         epochs.drop_log = drop_log
         epochs._bad_dropped = True
+    return epochs
+
+
+@verbose
+def read_epochs(fname, proj=True, add_eeg_ref=True, verbose=None):
+    """Read epochs from a fif file
+
+    Parameters
+    ----------
+    fname : str
+        The name of the file, which should end with -epo.fif or -epo.fif.gz.
+    proj : bool | 'delayed'
+        Apply SSP projection vectors. If proj is 'delayed' and reject is not
+        None the single epochs will be projected before the rejection
+        decision, but used in unprojected state if they are kept.
+        This way deciding which projection vectors are good can be postponed
+        to the evoked stage without resulting in lower epoch counts and
+        without producing results different from early SSP application
+        given comparable parameters. Note that in this case baselining,
+        detrending and temporal decimation will be postponed.
+        If proj is False no projections will be applied which is the
+        recommended value if SSPs are not used for cleaning the data.
+    add_eeg_ref : bool
+        If True, an EEG average reference will be added (unless one
+        already exists).
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+        Defaults to raw.verbose.
+
+    Returns
+    -------
+    epochs : instance of Epochs
+        The epochs
+    """
+    check_fname(fname, 'epochs', ('-epo.fif', '-epo.fif.gz'))
+
+    fnames = [fname]
+    epochs = list()
+    for fname in fnames:
+        logger.info('Reading %s ...' % fname)
+        fid, tree, _ = fiff_open(fname)
+        next_fname = _get_next_fname(fid, fname, tree)
+        epoch = _read_one_epoch_file(fid, tree, fname, proj, add_eeg_ref,
+                                     verbose)
+        epochs.append(epoch)
 
+        if next_fname is not None:
+            fnames.append(next_fname)
```

Comment:
I know some iterators throw a warning if you change the size of the object during iteration (maybe only on Py3k), I assume `list` isn't one of them?